twake/backend/node/src/services/previews/services/processing/service.ts

```diff
@@ -0,0 +1,34 @@
+import { generatePreview as thumbnailsFromImages } from "./image";
+import { convertFromOffice } from "./office";
+import { convertFromPdf } from "./pdf";
+import { isFileType } from "../../utils";
+import { pdfExtensions, officeExtensions, imageExtensions } from "./mime";
+import StorageAPI from "../../../../core/platform/services/storage/provider";
+import { PreviewPubsubRequest, ThumbnailResult } from "../../types";
+
+export class PreviewProcessService {
+  name: "PreviewProcessService";
+  version: "1";
+
+  constructor(readonly storage: StorageAPI) {}
+
+  async generateThumbnails(
+    document: Pick<PreviewPubsubRequest["document"], "filename" | "mime" | "path">,
+    options: PreviewPubsubRequest["output"],
+  ): Promise<ThumbnailResult[]> {
+    if (isFileType(document.mime, document.filename, officeExtensions)) {
+      const pdfPath = await convertFromOffice(document.path, options.pages);
+      const thumbnailPath = await convertFromPdf(pdfPath.output, options.pages);
+      return (await thumbnailsFromImages(thumbnailPath.output)).output;
+    }
+
+    if (isFileType(document.mime, document.filename, pdfExtensions)) {
+      const thumbnailPath = await convertFromPdf(document.path, options.pages);
+      return (await thumbnailsFromImages(thumbnailPath.output)).output;
+    }
+
+    if (isFileType(document.mime, document.filename, imageExtensions)) {
+      return (await thumbnailsFromImages([document.path])).output;
+    }
```

Comment:
```
    let pagesAsImage: string[] = [];

    if (isFileType(document.mime, document.filename, officeExtensions)) {
      const pdfPath = await convertFromOffice(document.path, options.pages);
      pagesAsImage = (await convertFromPdf(pdfPath.output, options.pages)).output;
    }

    if (isFileType(document.mime, document.filename, pdfExtensions)) {
      pagesAsImage = (await convertFromPdf(document.path, options.pages)).output;
    }

    if (isFileType(document.mime, document.filename, imageExtensions)) {
       pagesAsImage = [document.path];
    }

    return (await thumbnailsFromImages(images)).output;

```

Reply:
This is a proposal, if we can find better then let's do it if not lets do not change it.

Reply:
Christophe asked me to return in each situation